citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java

```diff
@@ -149,7 +150,27 @@ protected void handleExportRequest(HttpServletRequest req, HttpServletResponse r
 		
 		if(org.sakaiproject.citation.api.CitationService.RIS_FORMAT.equals(format))
 		{
-			String citationCollectionId = req.getParameter("citationCollectionId");
+			String citationCollectionId = null;
+			ContentResource resource = null;
+			try {
+				org.sakaiproject.content.api.ContentHostingService contentHostingService = (org.sakaiproject.content.api.ContentHostingService) ComponentManager.get(org.sakaiproject.content.api.ContentHostingService.class);
+				resource = contentHostingService.getResource(req.getParameter("resourceId"));
+				citationCollectionId = new String(resource.getContent());
+			}
+			catch (PermissionException e) {
+				org.sakaiproject.tool.api.SessionManager sessionManager = (org.sakaiproject.tool.api.SessionManager) ComponentManager.get(org.sakaiproject.tool.api.SessionManager.class);
```

Comment:
That's not possible here. It's not a servlet so there is no init() method to override. It's called CitationListAccesServlet but it implements an interface HttpAccess rather than extending HttpServlet (or something similar).